edx_lint/pylint/annotations_check.py

```diff
@@ -233,85 +249,126 @@ class AnnotationBaseChecker(BaseChecker):
     """
 
     # Override this in child classes
-    CONFIG_FILENAME = ""
+    CONFIG_FILENAMES = []
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        config_path = pkg_resources.resource_filename(
-            "code_annotations",
-            os.path.join("contrib", "config", self.CONFIG_FILENAME),
-        )
-        self.config = AnnotationConfig(config_path, verbosity=-1)
-        self.search = StaticSearch(self.config)
+        self.config_search = []
+        for config_filename in self.CONFIG_FILENAMES:
+            config_path = pkg_resources.resource_filename(
+                "code_annotations",
+                os.path.join("contrib", "config", config_filename),
+            )
+            config = AnnotationConfig(config_path, verbosity=-1)
+            search = StaticSearch(config)
+            self.config_search.append((config, search))
 
     def check_module(self, node):
         """
         Perform checks on all annotation groups for this module.
         """
-        # This is a hack to avoid re-creating AnnotationConfig every time
-        self.config.source_path = node.path[0]
-        results = self.search.search()
+        for config, search in self.config_search:
+            # This is a hack to avoid re-creating AnnotationConfig every time
+            config.source_path = node.path[0]
+            results = search.search()
 
-        current_annotations_group = []
-        for _file_name, results in results.items():
-            for current_annotations_group in self.search.iter_groups(results):
-                self.check_annotation_group(current_annotations_group, node)
+            current_annotations_group = []
+            for _file_name, results in results.items():
+                for current_annotations_group in search.iter_groups(results):
+                    self.check_annotation_group(search, current_annotations_group, node)
 
-    def check_annotation_group(self, annotations, node):
+    def check_annotation_group(self, search, annotations, node):
         raise NotImplementedError
 
 
+class CodeAnnotationChecker(AnnotationBaseChecker):
+    """
+    Run generic code annotation checks.
+
+    This makes use of code-annotations `check_results()` method. Elements from
+    `search.annotation_errors` are then parsed and exposed to pylint. Note that modifying the creation order of the
+    error types in code-annotations will lead to a modification of the numerical IDs of the errors here.
+
+    When creating a new annotation configuration, its filename should be added to
+    CodeAnnotationChecker.CONFIG_FILENAMES.
```

Comment:
Do these files need to live in code-annotations contrib/config?  Should we mention that, potentially with a link to https://github.com/edx/code-annotations/tree/master/code_annotations/contrib/config?